public/app/plugins/datasource/mssql/datasource.ts

```diff
@@ -1,191 +1,97 @@
-        return "'" + value.replace(/'/g, `''`) + "'";
-      } else {
-        return value;
-      }
-    }
-
-    if (typeof value === 'number') {
-      return value;
-    }
-
-    const quotedValues = _map(value, (val) => {
-      if (typeof value === 'number') {
-        return value;
-      }
-
-      return "'" + val.replace(/'/g, `''`) + "'";
-    });
-    return quotedValues.join(',');
+  getQueryModel(target?: SQLQuery, templateSrv?: TemplateSrv, scopedVars?: ScopedVars): MSSqlQueryModel {
+    return new MSSqlQueryModel(target, templateSrv, scopedVars);
   }
 
-  interpolateVariablesInQueries(
-    queries: MssqlQueryForInterpolation[],
-    scopedVars: ScopedVars
-  ): MssqlQueryForInterpolation[] {
-    let expandedQueries = queries;
-    if (queries && queries.length > 0) {
-      expandedQueries = queries.map((query) => {
-        const expandedQuery = {
-          ...query,
-          datasource: this.getRef(),
-          rawSql: this.templateSrv.replace(query.rawSql, scopedVars, this.interpolateVariable),
-          rawQuery: true,
-        };
-        return expandedQuery;
-      });
-    }
-    return expandedQueries;
+  getResponseParser(): ResponseParser {
+    return new MSSqlResponseParser();
   }
 
-  applyTemplateVariables(target: MssqlQuery, scopedVars: ScopedVars): Record<string, any> {
-    return {
-      refId: target.refId,
-      datasource: this.getRef(),
-      rawSql: this.templateSrv.replace(target.rawSql, scopedVars, this.interpolateVariable),
-      format: target.format,
-    };
+  async fetchDatasets(): Promise<string[]> {
+    const datasets = await this.runSql<{ name: string[] }>(showDatabases(), { refId: 'datasets' });
+    return datasets.fields.name.values.toArray().flat();
   }
 
-  async annotationQuery(options: any): Promise<AnnotationEvent[]> {
-    if (!options.annotation.rawQuery) {
-      return Promise.reject({ message: 'Query missing in annotation definition' });
-    }
-
-    const query = {
-      refId: options.annotation.name,
-      datasource: this.getRef(),
-      rawSql: this.templateSrv.replace(options.annotation.rawQuery, options.scopedVars, this.interpolateVariable),
-      format: 'table',
-    };
-
-    return lastValueFrom(
-      getBackendSrv()
-        .fetch<BackendDataSourceResponse>({
-          url: '/api/ds/query',
-          method: 'POST',
-          data: {
-            from: options.range.from.valueOf().toString(),
-            to: options.range.to.valueOf().toString(),
-            queries: [query],
-          },
-          requestId: options.annotation.name,
-        })
-        .pipe(
-          map(
-            async (res: FetchResponse<BackendDataSourceResponse>) =>
-              await this.responseParser.transformAnnotationResponse(options, res.data)
-          )
-        )
-    );
+  async fetchTables(dataset?: string): Promise<string[]> {
+    const tables = await this.runSql<{ name: string[] }>(showTables(dataset), { refId: 'tables' });
```

Comment:
I don't think we need to explicitly use the "name" prop here. Just use the first column values.

```
    const tables = await this.runSql<string[]>(buildTableQuery(dataset), { refId: 'tables' });
    return tables.map((t) => t[0]);
```

same with the other runsql calls

Reply:
the same as above.